src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1069,140 +1060,156 @@ private void WriteRedType(TreeType node)
+                            WriteLine("    : default;", field.Type);
+                            CloseBlock();
+                            CloseBlock();
                         }
                         else if (field.Type == "SyntaxNodeOrTokenList")
                         {
                             throw new InvalidOperationException("field cannot be a random SyntaxNodeOrTokenList");
                         }
+                        else if (i == 0)
+                        {
+                            WriteLine(" => GetRedAtZero(ref this.{0});", CamelCase(field.Name));
+                        }
                         else
                         {
-                            if (i == 0)
-                            {
-                                WriteLine("            return this.GetRedAtZero(ref this.{0});", CamelCase(field.Name));
-                            }
-                            else
-                            {
-                                WriteLine("            return this.GetRed(ref this.{0}, {1});", CamelCase(field.Name), i);
-                            }
+                            WriteLine(" => GetRed(ref this.{0}, {1});", CamelCase(field.Name), i);
                         }
-                        WriteLine("        }");
-                        WriteLine("    }");
                     }
                     WriteLine();
                 }
 
                 for (int i = 0, n = valueFields.Count; i < n; i++)
                 {
                     var field = valueFields[i];
-                    WriteComment(field.PropertyComment, "    ");
-                    WriteLine("    {0} {1}{2} {3} {{ get {{ return ((Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.{4})this.Green).{3}; }} }}",
-                        "public", OverrideOrNewModifier(field), field.Type, field.Name, node.Name
-                        );
+                    WriteComment(field.PropertyComment, "");
+                    WriteLine("{0} {1}{2} {3} => ((Syntax.InternalSyntax.{4})this.Green).{3};",
+                        "public", OverrideOrNewModifier(field), field.Type, field.Name, node.Name);
                     WriteLine();
                 }
 
                 //GetNodeSlot forces creation of a red node.
-                WriteLine("    internal override SyntaxNode GetNodeSlot(int index)");
-                WriteLine("    {");
-                WriteLine("        switch (index)");
-                WriteLine("        {");
-                for (int i = 0, n = nodeFields.Count; i < n; i++)
-                {
-                    var field = nodeFields[i];
+                Write("internal override SyntaxNode GetNodeSlot(int index)");
 
-                    if (field.Type != "SyntaxToken" && field.Type != "SyntaxList<SyntaxToken>")
+                if (!nodeFields.Where(f => f.Type != "SyntaxToken" && f.Type != "SyntaxList<SyntaxToken>").Any())
+                {
+                    WriteLine(" => null;");
```

Comment:
simplified cases where we generate a switch that *only* contained `index switch { _ => null; }` into just `=> null;`